cmd/livepeer_cli/wizard_ticketbroker.go

```diff
@@ -45,64 +46,34 @@ func (w *wizard) deposit() {
 		return
 	}
 
-	params, err := w.ticketBrokerParams()
-	if err != nil {
-		glog.Errorf("Error getting TicketBroker params: %v", err)
-		return
-	}
+	fmt.Printf("Current Deposit: %v\n", sender.Deposit)
+	fmt.Printf("Current Reserve: %v\n", sender.Reserve)
 
-	reqPenaltyEscrowAmount := big.NewInt(0)
-	if params.MinPenaltyEscrow.Cmp(sender.PenaltyEscrow) > 0 {
-		reqPenaltyEscrowAmount = new(big.Int).Sub(params.MinPenaltyEscrow, sender.PenaltyEscrow)
-	}
+	if sender.ReserveState == pm.Frozen {
+		currRound := w.currentRound()
 
-	if reqPenaltyEscrowAmount.Cmp(big.NewInt(0)) > 0 {
-		w.fundAndApproveSigners(sender.Deposit, sender.PenaltyEscrow, reqPenaltyEscrowAmount)
+		fmt.Printf("Current Round: %v\n", currRound)
+		fmt.Printf("Thaw Round: %v\n", sender.ThawRound)
+
+		fmt.Printf("Cannot deposit because sender's reserve is frozen and not yet thawed")
 		return
 	}
 
-	w.fundDeposit(sender.Deposit)
-
-	return
-}
-
-func (w *wizard) fundAndApproveSigners(currDeposit *big.Int, currPenaltyEscrow *big.Int, reqPenaltyEscrowAmount *big.Int) {
-	fmt.Printf("Current Deposit: %v\n", currDeposit)
-	fmt.Printf("Current Penalty Escrow: %v\n", currPenaltyEscrow)
-
-	fmt.Printf("Enter deposit amount in ETH (%v will be used cover the minimum penalty escrow) - ", eth.FormatUnits(reqPenaltyEscrowAmount, "ETH"))
-
-	amount := w.readPositiveFloatAndValidate(func(in float64) (float64, error) {
-		amountWei := eth.ToBaseUnit(big.NewFloat(in))
+	fmt.Printf("Enter deposit amount in ETH - ")
 
-		if amountWei.Cmp(reqPenaltyEscrowAmount) < 0 {
-			return 0, fmt.Errorf("must deposit at least %v to cover the minimum penalty escrow", eth.FormatUnits(reqPenaltyEscrowAmount, "ETH"))
-		}
+	depositAmount := w.readPositiveFloat()
 
-		return in, nil
-	})
+	fmt.Printf("Enter reserve amount in ETH - ")
 
-	amountWei := eth.ToBaseUnit(big.NewFloat(amount))
-	depositAmount := new(big.Int).Sub(amountWei, reqPenaltyEscrowAmount)
+	reserveAmount := w.readPositiveFloat()
 
 	form := url.Values{
-		"depositAmount":       {depositAmount.String()},
-		"penaltyEscrowAmount": {reqPenaltyEscrowAmount.String()},
+		"depositAmount": {eth.ToBaseUnit(big.NewFloat(depositAmount)).String()},
```

Comment:
Just curious, we don't have precision issues when handling potentially 18 decimal places of float here?

Reply:
There could be precision issues - I decided to stick with the current float64 helper usage in this PR for now and was thinking of coming back to re-consider helper usage separately (maybe switch to converting string -> `big.Float` instead of string -> float64)